tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/preloading/LSPUtils.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.ballerina.plugins.idea.preloading;
+
+import com.google.common.base.Strings;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.ui.MessageType;
+import io.ballerina.plugins.idea.extensions.BallerinaLSPExtensionManager;
+import io.ballerina.plugins.idea.notifiers.BallerinaAutoDetectNotifier;
+import io.ballerina.plugins.idea.sdk.BallerinaSdk;
+import io.ballerina.plugins.idea.sdk.BallerinaSdkUtils;
+import io.ballerina.plugins.idea.settings.autodetect.BallerinaAutoDetectionSettings;
+import io.ballerina.plugins.idea.settings.langserverlogs.LangServerLogsSettings;
+import org.jetbrains.annotations.NotNull;
+import org.wso2.lsp4intellij.IntellijLanguageClient;
+import org.wso2.lsp4intellij.client.languageserver.serverdefinition.ProcessBuilderServerDefinition;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.plugins.idea.BallerinaConstants.BAL_FILE_EXT;
+import static io.ballerina.plugins.idea.BallerinaConstants.LAUNCHER_SCRIPT_PATH;
+import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_EXPERIMENTAL;
+import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_LS_DEBUG;
+import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_LS_TRACE;
+
+/**
+ * Language server protocol related utils.
+ *
+ * @since 1.1.4
```

Comment:
Is this version correct?

Reply:
Yeah this is introduced with the 1.1.4 patch release